### PR TITLE
HDDS-8645. Change TestOzoneManagerHASnapshot to not subclass TestOzoneManagerHA

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
`TestOzoneManagerHASnapshot` intermittently fails because DB is getting closed due to Ratis's consistence state. 
```
Error:  org.apache.hadoop.ozone.om.TestOzoneManagerHASnapshot.testSnapshotNameConsistency  Time elapsed: 3.094 s  <<< ERROR!
java.lang.RuntimeException: java.io.IOException: Rocks Database is closed
	at org.apache.hadoop.ozone.om.TestOzoneManagerHASnapshot.lambda$testSnapshotNameConsistency$0(TestOzoneManagerHASnapshot.java:76)
	at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
	Suppressed: org.apache.ratis.util.ExitUtils$ExitException: Cannot load OM DB as it is in an inconsistent state.
		at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:141)
		at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:151)
		at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:155)
```

`TestOzoneManagerHASnapshot` inherits `TestOzoneManagerHA` and uses cluster created by the base class which is statically initialized by `BeforeAll`.  Although we don't run test parallelly, there is some possibility one test is interfering with other one and causing inconsistency.

This change is to create mini cluster for `TestOzoneManagerHASnapshot` instead of using inherited one.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8645

## How was this patch tested?
Ran this test 25 times in workflow and it passed all the time.
Run 1: https://github.com/hemantk-12/ozone/actions/runs/5019667607/jobs/9000405979
Run 2: https://github.com/hemantk-12/ozone/actions/runs/5019724655/jobs/9000519201

I tired to run 50 times too but it failed due to disk space: https://github.com/hemantk-12/ozone/actions/runs/5019375127/jobs/8999791082 

CI/CD passed in one go on fork
Run 1: https://github.com/hemantk-12/ozone/actions/runs/5019860313
Run 2: https://github.com/hemantk-12/ozone/actions/runs/5026615405

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java [hunk 1]
```diff
@@ -42,23 +51,44 @@
...
-public class TestOzoneManagerHASnapshot extends TestOzoneManagerHA {
+public class TestOzoneManagerHASnapshot {
+  private static MiniOzoneHAClusterImpl cluster;
+  private static OzoneClient client;
+  private static String volumeName;
+  private static String bucketName;
+  private static ObjectStore store;
+  private static OzoneBucket ozoneBucket;
+
+  @BeforeAll
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java:62]: Please add an `@AfterAll` method to close client and shutdown the cluster.  I think that will solve the out of space problem. ;)

**Review by adoroszlai — COMMENTED** [review]
Thanks @hemantk-12 for the fix and testing it in repeated runs.  With a post-test cleanup this should be fine.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java (hunk 1 above)
Comment (hemantk-12) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java:63]: Thanks for the catch. Fixed it.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java [hunk 2]
```diff
@@ -42,23 +52,51 @@
...
+    volumeName = ozoneBucket.getVolumeName();
+    bucketName = ozoneBucket.getName();
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java:90]: Please add
```suggestion
  @AfterAll
  public static void cleanUp() {
    IOUtils.closeQuietly(client);
    if (cluster != null) {
      cluster.shutdown();
    }
  }
```

and the necessary import:

```
import org.apache.hadoop.hdds.utils.IOUtils;
```

**Review by adoroszlai — APPROVED** [review]

**Comment by prashantpogde** [comment]
Thank you @hemantk-12 for making these changes.

**Pull request merged**